test/Microsoft.Framework.ConfigurationModel.Test/ConfigurationTest.cs

```diff
@@ -321,5 +322,16 @@ public void CanIterateAfterCastedToIEnumerable()
 
             Assert.Equal(3, srcCount);
         }
+
+        [Fact]
+        public void SetValueThrowsExceptionNoSourceRegistered()
+        {
+            var config = new Configuration();
+            var ex = Assert.Throws<InvalidOperationException>(() => config.Set("Title", "Welcome"));
+            Assert.Equal(
```

Comment:
This looks error prone, whenever we change the error messages, and if we end up localizing things. Consider using Data our Source properties of InvalidOperationException (i.e. setting it while throwing, and doing the comparison based on that.)